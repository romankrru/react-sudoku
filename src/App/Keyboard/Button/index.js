import React from "react";
import cn from "classnames";

import styles from "./index.module.css";

const Button = (props) => (
	<button
		className={cn(styles.button, {
			[styles.buttonBig]: props.size === "big",
		})}
		onClick={props.onClick}
		disabled={props.disabled}
	>
		{props.children}
	</button>
);

export default Button;
