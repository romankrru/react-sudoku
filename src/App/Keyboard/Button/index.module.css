.button {
    width: 55px;
    height: 50px;
	padding: 0;
	margin-right: 3%;

	font-size: 24px;
	line-height: 47px;
	text-align: center;

	background-color: #f7f9fc;
	border: 2px solid #dce1e8;
	border-radius: 10px;
	cursor: pointer;

	transition: border-color 0.2s ease;
}

.button:last-child {
	margin-right: 0;
}

.buttonBig {
	width: 240px;
}

.buttonDisabled {
	color: #dce1e8;

	cursor: default;
}

.button:hover {
	border-color: #c4c9d0;
}

.button:disabled:hover {
	border-color: #dce1e8;
}

.button:focus {
	outline: none;
}

.buttonCandidate {
	display: flex;
	padding: 6px;

	font-size: 16px;
	line-height: 1;
}

.buttonCandidate-1 { justify-content: flex-start; align-items: flex-start;  }
.buttonCandidate-2 { justify-content: center; align-items: flex-start;  }
.buttonCandidate-3 { justify-content: flex-end; align-items: flex-start;  }
.buttonCandidate-4 { justify-content: flex-start; align-items: center;  }
.buttonCandidate-5 { justify-content: center; align-items: center;  }
.buttonCandidate-6 { justify-content: flex-end; align-items: center;  }
.buttonCandidate-7 { justify-content: flex-start; align-items: flex-end;  }
.buttonCandidate-8 { justify-content: center; align-items: flex-end;  }
.buttonCandidate-9 { justify-content: flex-end; align-items: flex-end;  }

@media (min-width: 601px) {
	.keyboard {
		width: 60px;
		height: 60px;
	}
}
