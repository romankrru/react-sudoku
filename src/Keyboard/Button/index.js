import React from "react";
import cn from "classnames";

import styles from "./index.module.css";

const Button = (props) => {
	return (
		<button
			className={cn(styles.button, {
				[styles.buttonBig]: props.size === "big",
			})}
			disabled={props.disabled}
		>
			{props.children}
		</button>
	);
};

export default Button;
