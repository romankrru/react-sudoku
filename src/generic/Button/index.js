import React from "react";
import cn from "classnames";

import styles from "./index.module.css";

const Button = (props) => (
	<button className={cn(styles.button, props.className)}>
		{props.children}
	</button>
);

export default Button;
