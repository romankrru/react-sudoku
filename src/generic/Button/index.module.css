.button {
	padding: 6px 15px;

	font: inherit;
	line-height: 21px;
	color: white;
	text-align: center;

	background-color: transparent;
	border: 1px solid rgba(255, 255, 255, 0.4);
	border-radius: 10px;

	cursor: pointer;
	transition: all 0.2s ease;
}

.button:hover {
	background: rgba(255, 255, 255, 0.15);
}

.button:focus {
	outline: none;
}

.buttonBlue {
	color: #5392f6;

	border: 1px solid rgba(83, 146, 246, 0.6);
}

.buttonBlue:hover {
	border: 1px solid rgba(83, 146, 246, 1);
}
