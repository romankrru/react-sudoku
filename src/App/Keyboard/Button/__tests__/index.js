import React from "react";
import { render } from "@testing-library/react";

import Button from "../";

const renderButton = (props) => {
	const defaultProps = {
		onClick: jest.fn,
	};

	const view = render(<Button {...defaultProps} />);

	const rerender = (newProps) =>
		view.rerender(<Button {...{ ...defaultProps, ...newProps }} />);

	return { view, rerender };
};

describe("Keyboard <Button />", () => {
	it("renders without crashes", () => {
		const { view } = renderButton();
		expect(view.container).toMatchSnapshot();
	});

	it("handles styles correctly while in candidate mode", () => {
		const { view, rerender } = renderButton();

		expect(
			view.container.querySelector(".buttonCandidate.buttonCandidate-1")
		).not.toBeInTheDocument();

		rerender({ candidate: true, candidateValue: 1 });

		expect(
			view.container.querySelector(".buttonCandidate.buttonCandidate-1")
		).toBeInTheDocument();

		rerender({ candidate: true, candidateValue: 9 });

		expect(
			view.container.querySelector(".buttonCandidate.buttonCandidate-9")
		).toBeInTheDocument();

		rerender({ candidate: true, candidateValue: 3 });

		expect(
			view.container.querySelector(".buttonCandidate.buttonCandidate-3")
		).toBeInTheDocument();
	});
});
