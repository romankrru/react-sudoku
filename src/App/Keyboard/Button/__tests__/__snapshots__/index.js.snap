// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard <Button /> renders without crashes 1`] = `
<div>
  <div
    class="button"
  >
    <span />
  </div>
</div>
`;
